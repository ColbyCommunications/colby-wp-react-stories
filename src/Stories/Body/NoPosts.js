import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { withStateContext } from '../Context';

const StyledNoPosts = styled.div`
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
`;

const getNoPostsText = ({ searchTerm, fetching }) => {
  if (searchTerm.length) {
    return `No results for <i>${searchTerm}</i>`;
  }

  return fetching ? '' : 'No posts found.';
};

const NoPosts = ({ searchTerm, fetching }) => (
  <StyledNoPosts
    dangerouslySetInnerHTML={{
          __html: getNoPostsText({ searchTerm, fetching }),
        }}
  />
);

NoPosts.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  fetching: PropTypes.bool.isRequired,
};

export default withStateContext(NoPosts);
