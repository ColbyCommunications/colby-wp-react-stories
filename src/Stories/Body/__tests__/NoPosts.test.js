import React from 'react';
import renderer from 'react-test-renderer';

import { NoPosts } from '../NoPosts';

describe('NoPosts', () => {
  it('renders no results for search', () => {
    const tree = renderer.create(<NoPosts fetching={false} searchTerm="my search" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders no posts at all', () => {
    const tree = renderer.create(<NoPosts fetching={false} searchTerm="" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
