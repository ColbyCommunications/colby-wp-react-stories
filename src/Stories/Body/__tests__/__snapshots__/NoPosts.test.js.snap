// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPosts renders no posts at all 1`] = `
<div
  className="sc-bdVaJa kdnzwt"
  dangerouslySetInnerHTML={
    Object {
      "__html": "No posts found.",
    }
  }
/>
`;

exports[`NoPosts renders no results for search 1`] = `
<div
  className="sc-bdVaJa kdnzwt"
  dangerouslySetInnerHTML={
    Object {
      "__html": "No results for <i>my search</i>",
    }
  }
/>
`;
